TypeScript compile error `Type 'undefined' does not satisfy the constraint 'keyof NavigationInfo'.`

After updating to the latest of `navigation` (5.4.0) and `navigation-react` (4.2.0) I'm getting a Typescript error:

```
> tsc

node_modules/navigation/navigation.d.ts:567:55 - error TS2344: Type 'undefined' does not satisfy the constraint 'keyof NavigationInfo'.
  Type 'undefined' is not assignable to type 'string | number'.

567     parseLink<StateKey extends keyof NavigationInfo = undefined>(url: string): { state: StateKey extends undefined ? State : State<StateKey & string, NavigationInfo[StateKey]>; data: StateKey extends undefined ? any : NavigationInfo[StateKey]; crumbs: Crumb[] };
                                                          ~~~~~~~~~


Found 1 error.
```

I haven't taken a stab at trying to patch this on my end yet, any thoughts?

This is the tsconfig I'm compiling with:

```
{
  "compilerOptions": {
    "target": "ES2019",
    "module": "commonjs",
    "lib": ["ES2019"],
    "allowJs": true,
    "jsx": "react",
    "noEmit": true,
    "isolatedModules": true,
    "strict": true,
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "resolveJsonModule": true
  },
  "exclude": [
    "node_modules",
    "babel.config.js",
    "metro.config.js",
    "jest.config.js",
    "ios",
    "android"
  ]
}
```

--- a/Navigation/test/navigation-tests.ts
+++ b/Navigation/test/navigation-tests.ts
@@ -1,4 +1,4 @@
-// tsc --target es3 --lib ES2015,DOM --noImplicitAny true navigation-tests.ts
+// tsc --target es3 --lib ES2015,DOM --noImplicitAny true --strict true navigation-tests.ts
 import { HashHistoryManager, StateNavigator, State } from 'navigation';
 
 // History Manager

--- a/NavigationReact/test/navigation-react-tests.tsx
+++ b/NavigationReact/test/navigation-react-tests.tsx
@@ -1,4 +1,4 @@
-// tsc --jsx react --target es3 --lib ES2015,DOM --esModuleInterop --noImplicitAny true navigation-react-tests.tsx
+// tsc --jsx react --target es3 --lib ES2015,DOM --esModuleInterop --noImplicitAny true --strict true navigation-react-tests.tsx
 import { StateNavigator } from 'navigation';
 import { NavigationHandler, NavigationContext, NavigationEvent, NavigationBackLink, NavigationLink, RefreshLink } from 'navigation-react';
 import React, { useContext } from 'react';
@@ -31,7 +31,7 @@ const People = () => {
                 ))}
             </ul>
             <RefreshLink<AppNavigation, 'people'>
-                navigationData={{ page: page + 1 }}
+                navigationData={{ page: page! + 1 }}
                 disableActive={true}
                 includeCurrentData={true}>
                 Next

--- a/NavigationReactMobile/test/navigation-react-mobile-tests.tsx
+++ b/NavigationReactMobile/test/navigation-react-mobile-tests.tsx
@@ -1,4 +1,4 @@
-// tsc --jsx react --target es3 --lib ES2015,DOM --esModuleInterop --noImplicitAny true navigation-react-mobile-tests.tsx
+// tsc --jsx react --target es3 --lib ES2015,DOM --esModuleInterop --noImplicitAny true --strict true navigation-react-mobile-tests.tsx
 import { StateNavigator } from 'navigation';
 import { NavigationContext, NavigationEvent, NavigationLink } from 'navigation-react';
 import { NavigationMotion, MobileHistoryManager, SharedElement, SharedElementMotion } from 'navigation-react-mobile';

--- a/NavigationReactNative/test/navigation-react-native-tests.tsx
+++ b/NavigationReactNative/test/navigation-react-native-tests.tsx
@@ -1,4 +1,4 @@
-// tsc --jsx react --target es6 --lib ES2015 --esModuleInterop --noImplicitAny true navigation-react-native-tests.tsx
+// tsc --jsx react --target es6 --lib ES2015 --esModuleInterop --noImplicitAny true --strict true navigation-react-native-tests.tsx
 import React, { useContext, useState } from 'react';
 import { View, Text, ScrollView, TouchableHighlight } from 'react-native';
 import { StateNavigator } from 'navigation';

--- a/build/npm/navigation/navigation.d.ts
+++ b/build/npm/navigation/navigation.d.ts
@@ -563,8 +563,8 @@ export class StateNavigator<NavigationInfo extends { [index: string]: any } = an
      * Parses the url out into State and Navigation Data
      * @param url The url to parse
      */
-
-    parseLink<StateKey extends keyof NavigationInfo = undefined>(url: string): { state: StateKey extends undefined ? State : State<StateKey & string, NavigationInfo[StateKey]>; data: StateKey extends undefined ? any : NavigationInfo[StateKey]; crumbs: Crumb[] };
+    parseLink(url: string): { state: State; data: any; crumbs: Crumb[] };
+    parseLink<StateKey extends keyof NavigationInfo>(url: string): { state: State<StateKey & string, NavigationInfo[StateKey]>; data: NavigationInfo[StateKey]; crumbs: Crumb[] };
     /**
      * Creates a FluentNavigator
      * @param withContext a value indicating whether to inherit the current

